drivers/auxiliary/ups.cpp

```diff
@@ -0,0 +1,577 @@
+/*******************************************************************************
+  Copyright(c) 2025 Jérémie Klein. All rights reserved.
+
+  INDI UPS Driver using NUT (Network UPS Tools)
+  This driver monitors a UPS through NUT and exposes its status through INDI's weather interface.
+  Battery level and power status are mapped to weather parameters for compatibility.
+
+  This program is free software; you can redistribute it and/or modify it
+  under the terms of the GNU General Public License as published by the Free
+  Software Foundation; either version 2 of the License, or (at your option)
+  any later version.
+
+  This program is distributed in the hope that it will be useful, but WITHOUT
+  ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+  FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+  more details.
+
+  You should have received a copy of the GNU Library General Public License
+  along with this library; see the file COPYING.LIB.  If not, write to
+  the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+  Boston, MA 02110-1301, USA.
+
+  The full GNU General Public License is included in this distribution in the
+  file called LICENSE.
+*******************************************************************************/
+
+#include <memory>
+#include <chrono>
+#include <thread>
+#include <cstdio>
+#include <cstring>
+#include <unistd.h>
+#include <sstream>
+#include <termios.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <netdb.h>
+#include "ups.h"
+#include "indicom.h"
+
+// We declare an auto pointer to UPS
+std::unique_ptr<UPS> ups(new UPS());
+
+UPS::UPS()
+{
+    setVersion(1, 0);
+    setWeatherConnection(CONNECTION_NONE);
+    socketFd = -1;
+}
+
+const char *UPS::getDefaultName()
+{
+    return "UPS";
+}
+
+bool UPS::initProperties()
+{
+    INDI::Weather::initProperties();
+
+    // Setup server address properties
+    ServerAddressTP[0].fill("HOST", "Host", "localhost");
+    ServerAddressTP[1].fill("PORT", "Port", "3493");  // Default NUT port
+    ServerAddressTP.fill(getDeviceName(), "SERVER_ADDRESS", "NUT Server", CONNECTION_TAB, IP_RW, 60, IPS_IDLE);
+
+    // Setup UPS name property
+    UPSNameTP[0].fill("NAME", "UPS Name", "ups");  // Default UPS name
+    UPSNameTP.fill(getDeviceName(), "UPS_NAME", "UPS", CONNECTION_TAB, IP_RW, 60, IPS_IDLE);
+
+    // Setup connection settings properties
+    ConnectionSettingsNP[0].fill("TIMEOUT", "Timeout (sec)", "%.0f", 1, 30, 1, 5);
+    ConnectionSettingsNP[1].fill("RETRIES", "Max Retries", "%.0f", 1, 10, 1, 3);
+    ConnectionSettingsNP[2].fill("RETRY_DELAY", "Retry Delay (ms)", "%.0f", 100, 5000, 100, 1000);
+    ConnectionSettingsNP[3].fill("RECONNECT_ATTEMPTS", "Max Reconnect Attempts", "%.0f", 0, 10, 1, 3);
+    ConnectionSettingsNP.fill(getDeviceName(), "CONNECTION_SETTINGS", "Connection", CONNECTION_TAB, IP_RW, 60, IPS_IDLE);
```

Comment:
You can add this as well to load property from config.

`ConnectionSettingsNP.load();`

Reply:
I have done it in the loadConfig funciton.
It is not the right place ?